src/server/middleware/authentication.ts

```diff
@@ -0,0 +1,108 @@
+import passport from 'passport'
+import passportLocal, { VerifyFunctionWithRequest } from 'passport-local'
+import bcrypt from 'bcryptjs'
+import { Express } from 'express'
+import { FieldValidators, User, UserStatus, Validator } from '@openforis/arena-core'
+
+import { ExpressInitializer } from '../expressInitializer'
+import { UserServiceServer } from '../../service'
+import { UserResetPasswordRepository, AuthGroupRepository } from '../../repository'
+
+const comparePassword = bcrypt.compareSync
+
+// TODO: Handle these separately
+const ValidationMessages = {
+  emailInvalid: 'validationErrors.user.emailInvalid',
+  userNotFound: 'validationErrors.user.userNotFound',
+  passwordChangeRequired: 'validationErrors.user.passwordChangeRequired',
+}
+
+const _initializeUser = async (_user: User): Promise<User> => {
+  const user = { ..._user }
+  if (user.password) delete user.password
+
+  // Assoc auth groups
+  let userUpdated = {
+    ...user,
+    authGroups: await AuthGroupRepository.fetchUserGroups({ userUuid: user.uuid }),
+  }
+  if (user.status === UserStatus.INVITED) {
+    const expired = !(await UserResetPasswordRepository.existsResetPasswordValidByUserUuid({ userUuid: user.uuid }))
+    userUpdated = {
+      ...userUpdated,
+      invitation: {
+        expired,
+      },
+    }
+  }
+
+  // @ts-ignore fixed by implementing repository method
+  return userUpdated
+}
+
+const findUserByEmailAndPassword = async (email: string, password: string) => {
```

Comment:
Don't you need to specify the return type?

Reply:
The return type is inherited from `_initializeUser`

Reply:
better to be explicit